lib/arborist/build-ideal-tree.js

```diff
@@ -355,6 +359,41 @@ module.exports = cls => class IdealTreeBuilder extends cls {
     })
   }
 
+  async [_retrieveSpecName] (spec) {
+    // if it's just @'' then we reload whatever's there, or get latest
+    // if it's an explicit tag, we need to install that specific tag version
+    const isTag = spec.rawSpec && spec.type === 'tag'
+
+    if (spec.name && !isTag)
+      return spec
+
+    const mani = await pacote.manifest(spec)
+    // if it's a tag type, then we need to run it down to an actual version
+    if (isTag)
+      return npa(`${mani.name}@${mani.version}`)
+
+    spec.name = mani.name
+    return spec
+  }
+
+  async [_followSymlinkPath] (spec, currpath) {
+    if(spec.type === 'directory') {
+      const real = await (
+        realpath(spec.fetchSpec, this[_rpcache], this[_stcache])
+          // TODO: create synthetic test case to simulate realpath failure
+          .catch(/* istanbul ignore next */() => null)
+      )
+
+      /* istanbul ignore else - same as the ignored catch above */
+      if (real) {
+        const { name } = spec
+        spec = npa(`file:${relpath(this.path, real)}`, this.path)
```

Comment:
the new spec gets npa parsed relative to `this.path`, so that's what will end up in `resolved`, in the `package.json`, etc - also something I want to validate that makes sense in the impl 😊